pkg/oc/cli/cmd/registry/info/info.go

```diff
@@ -0,0 +1,188 @@
+		},
+	}
+
+	flag := cmd.Flags()
+	flag.BoolVar(&o.Check, "check", o.Check, "Attempt to contact the integrated registry.")
+	flag.BoolVarP(&o.Check, "quiet", "q", o.Quiet, "Suppress normal output and only print status.")
+	flag.BoolVar(&o.ShowInternal, "internal", o.ShowInternal, "Only check the internal registry hostname.")
+	flag.BoolVar(&o.ShowPublic, "public", o.ShowPublic, "Only check the public registry hostname.")
+
+	return cmd
+}
+
+func (o *Options) Complete(f *clientcmd.Factory, args []string) error {
+	cfg, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+	client, err := imageclient.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	o.Client = client
+
+	ns, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Namespaces = []string{ns, "openshift"}
+
+	return nil
+}
+
+type RegistryInfo struct {
+	Public   string
+	Internal string
+}
+
+func (i *RegistryInfo) Installed() bool {
+	return len(i.Public) > 0 || len(i.Internal) > 0
+}
+
+func (i *RegistryInfo) HostPort() (string, bool) {
+	if len(i.Public) > 0 {
+		return i.Public, true
+	}
+	return i.Internal, false
+}
+
+func findRegistryInfo(client imageclient.Interface, namespaces ...string) (*RegistryInfo, error) {
+	for _, ns := range namespaces {
+		imageStreams, err := client.Image().ImageStreams(ns).List(metav1.ListOptions{})
+		if err != nil || len(imageStreams.Items) == 0 {
+			continue
+		}
+		is := imageStreams.Items[0]
+
+		info := &RegistryInfo{}
+		if value := is.Status.PublicDockerImageRepository; len(value) > 0 {
+			ref, err := imageapi.ParseDockerImageReference(value)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse public registry info from the server")
+			}
+			info.Public = ref.Registry
+		}
+		if value := is.Status.DockerImageRepository; len(value) > 0 {
+			ref, err := imageapi.ParseDockerImageReference(value)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse internal registry info from the server")
+			}
+			info.Internal = ref.Registry
+		}
+		if !info.Installed() {
+			return nil, fmt.Errorf("the integrated registry has not been configured")
+		}
+		return info, nil
+	}
+	return nil, fmt.Errorf("no image streams could be located to retrieve registry info, please specify a namespace with image streams")
+}
+
+func (o *Options) Validate() error {
+	if o.ShowInternal && o.ShowPublic {
+		return fmt.Errorf("only one of --internal or --public may be specified at a time")
+	}
+	return nil
+}
+
+func (o *Options) Run() error {
+	info, err := findRegistryInfo(o.Client, o.Namespaces...)
+	if err != nil {
+		return err
+	}
+
+	var host string
+	var public bool
+	switch {
+	case o.ShowPublic:
+		host = info.Public
+	case o.ShowInternal:
+		host = info.Internal
+		public = false
+	default:
+		host, public = info.HostPort()
+	}
+
+	if o.Check {
+		ctx := apirequest.NewContext()
+		if !public && !o.ShowInternal {
+			fmt.Fprintf(o.ErrOut, "info: Registry does not have a public hostname registered\n")
```

Comment:
s/info/warning

Reply:
It's not a warning, you can run this inside a cluster successfully.